### DISCUSSION TITLE
My GLSL shaders.

### DISCUSSION THREAD

**Discussion by MarioMey** [Limeth/obs-shaderfilter-plus#20] [Show and tell]
I'm on Linux, so I use ShaderFilter-Plus in GLSL. With some help of Limeth, I could transpose some shaders that I used in Blender Game Engine to this plugin. And some others, one from ShaderToy (Warp) and another new from scratch (Wave). I want to share them with you ♥.

All of them: https://github.com/MarioMey/obs-shaders
## Blur
https://github.com/MarioMey/obs-shaders/blob/main/blur.glsl
![Blur](https://user-images.githubusercontent.com/4954109/119679247-95664e80-be16-11eb-9527-79bfca102379.png)
## Chromatic Aberration
https://github.com/MarioMey/obs-shaders/blob/main/chromatic_aberration.glsl
![Chromatic Aberration](https://user-images.githubusercontent.com/4954109/119679464-c0e93900-be16-11eb-9f77-5698e2944a19.png)
## Upwards_distorted (fork of NetherEran shader)
https://github.com/MarioMey/obs-shaders/blob/main/quad_don_gilberto_manhattan_ruiz.glsl
![Don Gilberto Manhattan Ruiz Cuadrático](https://user-images.githubusercontent.com/4954109/119679893-1a516800-be17-11eb-89e0-ecd4fdbdaccf.png)
### Distort
https://github.com/MarioMey/obs-shaders/blob/main/distort.glsl
![imagen](https://user-images.githubusercontent.com/4954109/122564719-0ebf1e80-d01c-11eb-8e77-29dadb58eb31.png)
## Noise
https://github.com/MarioMey/obs-shaders/blob/main/noise.glsl
![Noise](https://user-images.githubusercontent.com/4954109/119679543-d3637280-be16-11eb-86db-7b2c6d129c03.png)
### Lens distortion
https://github.com/MarioMey/obs-shaders/blob/main/lens_distortion.glsl
![imagen](https://user-images.githubusercontent.com/4954109/122564788-24344880-d01c-11eb-9349-9efd50a73ce8.png)
## Pixelate
https://github.com/MarioMey/obs-shaders/blob/main/pixelate.glsl
![Pixelate](https://user-images.githubusercontent.com/4954109/119679739-f9891280-be16-11eb-866c-6f402f1c676d.png)
## Radial Blur + Zoom (the best!)
https://github.com/MarioMey/obs-shaders/blob/main/radialBlur.glsl
![Radial Blur](https://user-images.githubusercontent.com/4954109/119680269-67cdd500-be17-11eb-82c1-3f8d079fabfc.png)
## Wave
https://github.com/MarioMey/obs-shaders/blob/main/wave.glsl
![Wave](https://user-images.githubusercontent.com/4954109/119680176-51277e00-be17-11eb-8b01-8f35ef090cd5.png)

**Comment by Limeth** [comment]
Great stuff, thank you very much for sharing! :+1:

**Comment by MarioMey** [comment]
New ones:
* Distort
* Lens distortion.

Updated:
* RadialBlur, now it has a Zoom option.

Removed:
* Warp

**Comment by nothingxs** [comment]
I am getting an error using pixelate.glsl, which produces the following:

> Could not create the effect due to the following error:
> device_pixelshader_create (D3D11): Compiler warnings/errors for C:/Program Files/obs-studio/data/obs-plugins/obs-shaderfilter-plus/shaders/pixelate.glsl (Pixel shader, technique Draw, pass 0):
> C:\Program Files\obs-studio\data\obs-plugins\obs-shaderfilter-plus\shaders\pixelate.glsl (Pixel shader, technique Draw, pass 0)(13,1-4): error X3000: unrecognized identifier 'vec4'
> 
> Pass (0) <> missing pixel shader!

Any ideas?

**Reply by MarioMey** [reply]
Sorry, I don't know... all other shaders do work?

**Reply by Limeth** [reply]
I believe @MarioMey is using the OpenGL/GLSL graphics API and his shaders are written in GLSL, while this error seems to indicate that you are using the DirectX/HLSL API instead. Unfortunately, OBS does not implement any automatic conversion from GLSL to HLSL. There are two ways to get these shaders working in DirectX/HLSL:
1. Rewrite the shaders to HLSL. Usually, this requires just replacing some keywords like `vec4` to `float4`, and so on, according to: https://docs.microsoft.com/en-us/windows/uwp/gaming/glsl-to-hlsl-reference
2. Force the usage of OpenGL/GLSL on Windows, as [described in the README](https://github.com/Limeth/obs-shaderfilter-plus#cross-platform-glsl). However, this usually results in lower performance compared to DirectX/HLSL on Windows.
